src/main/java/seedu/address/model/task/CompletionStatus.java

```diff
@@ -1,33 +1,47 @@
 package seedu.address.model.task;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
 
 /**
  * Represents a Task's completion status in the task list.
  * Guarantees: immutable;
  */
 public class CompletionStatus {
-    public final boolean value;
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Completion status must be either true or false";
+    public final String value;
+    private final Boolean booleanValue;
 
     /**
      * Constructs an {@code CompletionStatus}.
      *
      * @param completionStatus A valid completion status.
      */
-    public CompletionStatus(Boolean completionStatus) {
+    public CompletionStatus(String completionStatus) {
         requireNonNull(completionStatus);
+        checkArgument(isValidCompletionStatus(completionStatus), MESSAGE_CONSTRAINTS);
         value = completionStatus;
+        booleanValue = Boolean.parseBoolean(completionStatus);
+    }
+
+    /**
+     * Returns true if a given string is a valid CompletionStatus.
+     */
+    public static boolean isValidCompletionStatus(String test) {
+        return ("true".equals(test) || "false".equals(test));
```

Comment:
Would it be better to keep it as a `Boolean` instead of a `String`? As discussed, lets open it as an issue for now and leave it as a `String` for now to ensure that there will be fewer problems when we merge `V1.2-Evolution`!

Reply:
Will make an issue to track it.